### PR TITLE
[omemo] Added recipes needed to build omemo module

### PR DESCRIPTION
**Pull request by goffi-contrib**
OMEMO is an end to end encryption algorithm used in XMPP. This recipe
add the module implementing it + its dependencies.
The following dependencies have been added:
        - DoubleRatchet
        - PyNaCl,
        - X3DH
        - XEdDSA

This has been tested working with python 2.

### CODE REVIEW

File: pythonforandroid/recipes/doubleratchet/__init__.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from pythonforandroid.recipe import PythonRecipe
+
+
+class DoubleRatchetRecipe(PythonRecipe):
+    name = 'doubleratchet'
+    version = '0.4.0'
+    url = 'https://files.pythonhosted.org/packages/2c/50/4f8caeb11b211879937c77dbe08db6b17e2253ee296fedda548df7938093/DoubleRatchet-0.4.0.tar.gz'
```
Comment (AndreMiras) [inline pythonforandroid/recipes/doubleratchet/__init__.py]: url should be:
```python
url = 'https://pypi.python.org/packages/source/D/DoubleRatchet/DoubleRatchet-{version}.tar.gz'
```

**Comment by goffi-contrib** [comment]
@AndreMiras URLs are fixed.

**Comment by AndreMiras** [comment]
Thanks it's looking good.
We need https://github.com/kivy/python-for-android/pull/1437 to be merged before we can merge yours.
I'll keep you posted later during the coming week

**Comment by AndreMiras** [comment]
@goffi-contrib https://github.com/kivy/python-for-android/pull/1437 just got merged. Do you mind rebasing to see if Travis is now happy about it?

**Comment by goffi-contrib** [comment]
@AndreMiras done :)

**Comment by goffi-contrib** [comment]
@AndreMiras still the same missing autoreconf issue: `/home/user/.local/share/python-for-android/build/other_builds/libffi/armeabi/libffi/autogen.sh: 2: exec: autoreconf: not found`

**Comment by AndreMiras** [comment]
Yes basically everything that relies on `libffi` would fail here :disappointed:

File: ci/constants.py [hunk 2]
```diff
@@ -38,6 +38,13 @@ class TargetPython(Enum):
     'm2crypto',
     'netifaces',
     'Pillow',
+    # depends on cffi that still seems to have compilation issues
+    'protobuf_cpp',
+    'xeddsa',
+    'x3dh',
+    'pynacl',
+    'doubleratchet',
+    'omemo',
```
Comment (goffi-contrib) [inline ci/constants.py]: @AndreMiras now that you have added autotools to the docker image, I guess I can remove those lines, right?

Comment (AndreMiras) [reply ci/constants.py]: Yes good remark, but well, it's complicated :joy: 
The libffi is fixed, but cffi still seems to fail for some reason. I didn't get time to investigate and I didn't want to postpone your merge further more.

**Pull request merged**